### PR TITLE
is this what you want?

### PR DESCRIPTION
**Pull request by ModProg**

### CODE REVIEW

File: sysfs/sysfs_macros/src/lib.rs [hunk 1]
```diff
@@ -388,7 +388,8 @@ struct SetterFunction {
 fn let_sysfs_path(sysfs_dir: &Option<LitStr>, sysfs_file: &str) -> Stmt {
     let literal = match sysfs_dir {
         Some(sysfs_dir) => format!("{}/{}", sysfs_dir.value(), sysfs_file),
-        None => format!("{}/{}", "{SYSFS_DIR}", sysfs_file),
+        // TODO what is SYSFS_DIR, where is it supposed to come from?
+        None => format!("{}/{}", "SYSFS_DIR", sysfs_file),
```
Comment (ModProg) [inline sysfs/sysfs_macros/src/lib.rs:392]: no idea what constant `SYSFS_DIR` is supposed to reference, did a grep and seems to just not exist.

File: sysfs/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,7 @@
 pub use sysfs_lib as lib;
 pub use sysfs_lib::{Error, Result};
 
+extern crate self as sysfs;
```
Comment (spikespaz) [inline sysfs/src/lib.rs:4]: What does this accomplish?

File: sysfs/sysfs_macros/src/lib.rs (hunk 1 above)
Comment (spikespaz) [reply sysfs/sysfs_macros/src/lib.rs:392]: Formatting `"{}", "{SYSFS_DIR}"` is intentional. This is a super lazy way to get some feedback from the compiler. The macro can make use of `static SYSFS_DIR: &str` to make a default path from it if unspecified in the attribute.

Interpolating a format segment into the literal means that the macro output will directly include the text `{SYSFS_DIR}` which can then be substituted if the symbol is defined outside of the macro invocation.

File: sysfs/src/lib.rs (hunk 2 above)
Comment (ModProg) [reply sysfs/src/lib.rs:4]: it makes `sysfs` available inside `sysfs`.

Problem is, I'm not sure I fully understood your question, so I just tried fixing the compilation errors.

File: sysfs/sysfs_macros/src/lib.rs (hunk 1 above)
Comment (ModProg) [reply sysfs/sysfs_macros/src/lib.rs:392]: oh ok, I was just confused because your usages didn't do that.

File: sysfs/src/lib.rs (hunk 2 above)
Comment (spikespaz) [reply sysfs/src/lib.rs:4]: Well this branch was my first attempt, or at least, I think it was but I got it out of a random stash. The question was really open ended which is why you and togglebit had trouble with me. I kinda just wanted to make sure that I set up the workspace correctly. I wanted two nested workspaces, but having a package-workspace inside of a workspace is not allowed. So I guess I'll just keep what I have on master with the two top level crates completely separated.

**Comment by spikespaz** [comment]
I deleted the branch because it was significantly behind `master`, and `master` included the changes from this PR. The only unresolved thing is to eliminate `sysfs_lib`.

**Pull request closed without merging**